test/test-filter-process.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+. "test/testlib.sh"
+
+if [ "$GIT_LFS_USE_LEGACY_FILTER" == "1" ]; then
+  echo "skip: $0 (filter process disabled)"
+  exit
+fi
```

Comment:
If we check the protocol filter capability via version number below - do we still need the variable check here?